aries_cloudagent/protocols/present_proof/v2_0/formats/dif/handler.py

```diff
@@ -186,47 +185,111 @@ async def create_pres(
             challenge = str(uuid4())
 
         input_descriptors = pres_definition.input_descriptors
+        claim_fmt = pres_definition.fmt
+        dif_handler_proof_type = None
         try:
             holder = self._profile.inject(VCHolder)
             record_ids = set()
             credentials_list = []
             for input_descriptor in input_descriptors:
-                expanded_types = set()
-                schema_ids = set()
+                tag_query = {"$and": []}
+                proof_type = None
+                limit_disclosure = input_descriptor.constraint.limit_disclosure and (
+                    input_descriptor.constraint.limit_disclosure == "required"
+                )
                 for schema in input_descriptor.schemas:
+                    tag_query_or_list = []
                     uri = schema.uri
-                    required = schema.required or True
+                    if schema.required is None:
+                        required = True
+                    else:
+                        required = schema.required
                     if required:
```

Comment:
As discussed, I think this tag filtering belongs in VCHolder so that it can be specialized for the specific backends, but it's fine to merge this as-is and I can revisit it later with the shared-components work. Most important is that it works for now.